### PR TITLE
fix: issues 269 and 270

### PR DESCRIPTION
**Pull request by bhumilsarvaiya**
## Summary of change

- The json body parser will additionally be used to parse the request body if the request body is of type `{}`
- `init` function will throw error if there are empty items in recipeList when passing the config

## Related issues

-   #269
-   #270

## Test Plan
- add a config test to test the scenario where an empty item is passed in recipeList array.

## Checklist for important updates

-   [x] Changelog has been updated
-   [x] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [x] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [x] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
-   [x] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [x] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [x] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

### CODE REVIEW

File: lib/ts/framework/utils.ts [hunk 1]
```diff
@@ -127,7 +127,12 @@ export async function assertThatBodyParserHasBeenUsedForExpressLikeRequest(
                     });
                 }
             }
-        } else if (request.body === undefined || Buffer.isBuffer(request.body)) {
+        } else if (
+            request.body === undefined ||
+            Buffer.isBuffer(request.body) ||
+            Object.keys(request.body).length === 0
+        ) {
```
Comment (bhumilsarvaiya) [inline lib/ts/framework/utils.ts:134]: there was issue while testing with next.js app. added the `setImmediate` fix

File: test/thirdpartyemailpassword/signinFeature.test.js [hunk 2]
```diff
@@ -303,6 +303,172 @@ describe(`signinFeature: ${printPath("[test/thirdpartyemailpassword/signinFeatur
...
+                })
+                .expect(200)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(JSON.parse(res.text).user);
+                    }
+                })
+        );
```
[the remark this answers is not in the document]
Comment (bhumilsarvaiya) [reply test/thirdpartyemailpassword/signinFeature.test.js:367]: done

**Review by rishabhpoddar — APPROVED** [review]
